### PR TITLE
LEB-4401: Only add preview image if it exists

### PR DESCRIPTION
**Pull request by kaynenh**
Only add preview image if it exists, otherwise still export entity without silently failing...

### CODE REVIEW

File: modules/acquia_lift_publisher/src/EventSubscriber/Cdf/EntityRenderHandler.php [hunk 1]
```diff
@@ -183,12 +183,10 @@ public function onCreateCdf(CreateCdfEntityEvent $event) {
           if (isset($view_modes['acquia_lift_preview_image'])) {
             $preview_image = $entity->{$view_modes['acquia_lift_preview_image']}->first();
 
-            if (!$preview_image) {
-              continue;
+            if ($preview_image) {
+              $src = ImageStyle::load('acquia_lift_publisher_preview_image')->buildUrl($preview_image->entity->getFileUri());
```
Comment (scor) [inline modules/acquia_lift_publisher/src/EventSubscriber/Cdf/EntityRenderHandler.php]: Looks good. I wonder if we should also add another level of conditional to make sure that the imagestyle can actually build a valid URL as opposed to an empty string. Not sure how confident we can be that given an entity, buildUrl() will work.

Comment (scor) [reply modules/acquia_lift_publisher/src/EventSubscriber/Cdf/EntityRenderHandler.php]: in other words, not add the `preview_image` attribute to cdf if we have an empty `$src`. Let me know what you think and I can update the PR as well.

Comment (kaynenh) [reply modules/acquia_lift_publisher/src/EventSubscriber/Cdf/EntityRenderHandler.php]: Noted. I came up with something a bit different to hopefully protect further.

**Comment by kaynenh** [comment]
Also note, I did not test this so that should probably happen before merging.

**Comment by scor** [comment]
Thanks for all your help @kaynenh - I'll test the code soon.

**Comment by kaynenh** [comment]
Yep! I think the scenario is if you associate a preview image to a bundle but then do not actually add an image to an entity of that bundle, it should still export with this code.

**Comment by scor** [comment]
We will want to wait for #291 to get merged before merging this one as they are both touching the same lines. #291 is pretty large and blocked by some failing tests right now. It should be easy to re-apply the changes here on top of it though once it's merged.

**Comment by scor** [comment]
Moved to #304 in dedicated branch. let's continue the conversation there.

**Pull request closed without merging**